File: 001 - Plus Minus/Solution.py
#!/bin/python3

import math
import os
import random
import re
import sys

#
# Complete the 'plusMinus' function below.
#
# The function accepts INTEGER_ARRAY arr as parameter.
#

def plusMinus(arr):
    # Write your code here
    counter = [0,0,0]

    for value in arr:
        if value > 0:
            counter[0] += 1
        elif value < 0:
            counter[1] += 1
        else:
            counter[2] += 1
    
    print(round(counter[0]/len(arr), 6))
    print(round(counter[1]/len(arr), 6))
    print(round(counter[2]/len(arr), 6))

if __name__ == '__main__':
    n = int(input().strip())

    arr = list(map(int, input().rstrip().split()))

    plusMinus(arr)

File: 001 - Plus Minus/Solution.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.function.*;
import java.util.regex.*;
import java.util.stream.*;
import static java.util.stream.Collectors.joining;
import static java.util.stream.Collectors.toList;

class Result {

    /*
     * Complete the 'plusMinus' function below.
     *
     * The function accepts INTEGER_ARRAY arr as parameter.
     */

    public static void plusMinus(List<Integer> arr) {
    // Write your code here
        double positivos = 0;
        double negativos = 0;
        double zeros = 0;

        for (Integer i : arr) {
            if (i > 0) {
                positivos++;
            } else if (i < 0) {
                negativos++;
            } else {
                zeros++;
            }
        }

        System.out.println(round(positivos/arr.size(), 6));
        System.out.println(round(negativos/arr.size(), 6));
        System.out.println(round(zeros/arr.size(), 6));
        
    }

    private static double round(double value, int places) {
        if (places < 0) throw new IllegalArgumentException();
    
        BigDecimal bd = new BigDecimal(Double.toString(value));
        bd = bd.setScale(places, RoundingMode.HALF_UP);
        return bd.doubleValue();
    }
    

}

public class Solution {
    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(bufferedReader.readLine().trim());

        List<Integer> arr = Stream.of(bufferedReader.readLine().replaceAll("\\s+$", "").split(" "))
            .map(Integer::parseInt)
            .collect(toList());

        Result.plusMinus(arr);

        bufferedReader.close();
    }
}
